aws-ec2-launch-configuration-block-device-unencrypted.yaml

```diff
@@ -0,0 +1,60 @@
+rules:
+- id: aws-ec2-launch-configuration-block-device-unencrypted
+  pattern-either:
+  - patterns:
+    - pattern: |
+        resource "aws_launch_configuration" $ANYTHING {
+          ...
+        }
+    - pattern-not-inside: |
+        resource "aws_launch_configuration" $ANYTHING {
+          ...
+          root_block_device {
+            ...
+            encrypted = true
+            ...
+          }
+          ...
+        }
+  - patterns:
+    - pattern-inside: |
+        resource "aws_launch_configuration" $ANYTHING {
+          ...
+        }
+    - pattern: |
+        ebs_block_device {
+          ...
+        }
+    - pattern-not: |
+        ebs_block_device {
+          ...
+          encrypted = true
+          ...
+        }
+  message: The AWS launch configuration block device is unencrypted. The block device
```

Comment:
My rule **does** use  the yaml operator `>-`:

<img width="594" alt="Screenshot 2023-02-23 at 3 30 06 PM" src="https://user-images.githubusercontent.com/5559014/221022972-1ad284b1-fd67-477f-a6fa-73fe2252b647.png">

I don't know why that disappears when I publish it from the semgrep app?

Reply:
Sadly the yaml parsing I believe in semgrep-app removes this if you open a rule from the registry it goes to this even if its >-